diff --git a/drivers/gpu/drm/panel/panel-newvision-nv3051d.c b/drivers/gpu/drm/panel/panel-newvision-nv3051d.c
index 2256038..850f293 100644
--- a/drivers/gpu/drm/panel/panel-newvision-nv3051d.c
+++ b/drivers/gpu/drm/panel/panel-newvision-nv3051d.c
@@ -474,14 +474,14 @@ static const struct drm_display_mode nv3051d_rgxx3_modes[] = {
 static const struct drm_display_mode nv3051d_rk2023_modes[] = {
 	{
 		.hdisplay       = 640,
-		.hsync_start    = 640 + 40,
-		.hsync_end      = 640 + 40 + 2,
-		.htotal         = 640 + 40 + 2 + 80,
+		.hsync_start    = 640 + 48,
+		.hsync_end      = 640 + 48 + 2,
+		.htotal         = 640 + 48 + 2 + 47,
 		.vdisplay       = 480,
-		.vsync_start    = 480 + 18,
-		.vsync_end      = 480 + 18 + 2,
-		.vtotal         = 480 + 18 + 2 + 4,
-		.clock          = 24150,
+		.vsync_start    = 480 + 2,
+		.vsync_end      = 480 + 2 + 4,
+		.vtotal         = 480 + 2 + 4 + 3,
+		.clock          = 21600,
 		.flags          = DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC,
 	},
 };
@@ -514,7 +514,8 @@ static const struct nv3051d_panel_info nv3051d_rk2023_info = {
 	.height_mm = 57,
 	.bus_flags = DRM_BUS_FLAG_DE_LOW | DRM_BUS_FLAG_PIXDATA_DRIVE_NEGEDGE,
 	.mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
-		      MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_NO_EOT_PACKET,
+		      MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_NO_EOT_PACKET |
+		      MIPI_DSI_CLOCK_NON_CONTINUOUS,
 };
 
 static const struct of_device_id newvision_nv3051d_of_match[] = {
