diff --git a/drivers/usb/dwc2/platform.c b/drivers/usb/dwc2/platform.c
index ef0d730..44dffd9 100644
--- a/drivers/usb/dwc2/platform.c
+++ b/drivers/usb/dwc2/platform.c
@@ -655,6 +655,9 @@ static int __maybe_unused dwc2_suspend(struct device *dev)
 	if (!dwc2->ll_hw_enabled)
 		return 0;
 
+	if (dwc2->core_reset_failed)
+		return 0;
+
 	is_device_mode = dwc2_is_device_mode(dwc2);
 	if (is_device_mode)
 		dwc2_hsotg_suspend(dwc2);
@@ -735,6 +738,9 @@ static int __maybe_unused dwc2_resume(struct device *dev)
 	if (!dwc2->ll_hw_enabled)
 		return 0;
 
+	if (dwc2->core_reset_failed)
+		return 0;
+
 	if (dwc2->phy_off_for_suspend && dwc2->ll_hw_enabled) {
 		ret = __dwc2_lowlevel_hw_enable(dwc2);
 		if (ret)
