diff --git a/include/linux/of_gpio.h b/include/linux/of_gpio.h
index d0f66a5e1..06fd55fc5 100644
--- a/include/linux/of_gpio.h
+++ b/include/linux/of_gpio.h
@@ -17,8 +17,26 @@
 
 struct device_node;
 
+enum of_gpio_flags {
+	OF_GPIO_ACTIVE_LOW = 0x1,
+	OF_GPIO_SINGLE_ENDED = 0x2,
+	OF_GPIO_OPEN_DRAIN = 0x4,
+	OF_GPIO_TRANSITORY = 0x8,
+	OF_GPIO_PULL_UP = 0x10,
+	OF_GPIO_PULL_DOWN = 0x20,
+	OF_GPIO_PULL_DISABLE = 0x40,
+};
+
 #ifdef CONFIG_OF_GPIO
 
+extern int of_get_named_gpio_flags(const struct device_node *np,
+	const char *list_name, int index, enum of_gpio_flags *flags);
+static inline int of_get_gpio_flags(const struct device_node *np, int index,
+	enum of_gpio_flags *flags)
+{
+	return of_get_named_gpio_flags(np, "gpios", index, flags);
+}
+
 extern int of_get_named_gpio(const struct device_node *np,
 			     const char *list_name, int index);
 
