diff --git a/drivers/gpio/gpiolib-of.c b/drivers/gpio/gpiolib-of.c
index 7bb33c467..969fed09f 100644
--- a/drivers/gpio/gpiolib-of.c
+++ b/drivers/gpio/gpiolib-of.c
@@ -30,15 +30,7 @@
  * match, but GPIO controllers are free to translate their own flags to
  * Linux-specific in their .xlate callback. Though, 1:1 mapping is recommended.
  */
-enum of_gpio_flags {
-	OF_GPIO_ACTIVE_LOW = 0x1,
-	OF_GPIO_SINGLE_ENDED = 0x2,
-	OF_GPIO_OPEN_DRAIN = 0x4,
-	OF_GPIO_TRANSITORY = 0x8,
-	OF_GPIO_PULL_UP = 0x10,
-	OF_GPIO_PULL_DOWN = 0x20,
-	OF_GPIO_PULL_DISABLE = 0x40,
-};
+
 
 /**
  * of_gpio_named_count() - Count GPIOs for a device
@@ -446,6 +438,14 @@ static struct gpio_desc *of_get_named_gpiod_flags(const struct device_node *np,
 	return desc;
 }
 
+int of_get_named_gpio_flags(const struct device_node *np, const char *name,
+	int index, enum of_gpio_flags *flags)
+{
+	struct gpio_desc *desc = of_get_named_gpiod_flags(np, name, index, flags);
+	return IS_ERR(desc) ? PTR_ERR(desc) : desc_to_gpio(desc);
+}
+EXPORT_SYMBOL_GPL(of_get_named_gpio_flags);
+
 /**
  * of_get_named_gpio() - Get a GPIO number to use with GPIO API
  * @np:		device node to get GPIO from
