--- a/include/configs/odroidgoa.h
+++ b/include/configs/odroidgoa.h
@@ -82,12 +82,10 @@
 		"console=ttyFIQ0,115200n8 "				\
 		"rw root=/dev/mmcblk0p2 rootwait rw fsck.repair=yes "	\
 		"net.iframes=0 fbcon=rotate:3\0"	\
-	"bootcmd=mmc dev 1; cfgload; run setbootargs;"	\
-		"load mmc 1:1 0x02000000 Image; "		\
-		"load mmc 1:1 0x01f00000 ${dtb_name}; "	\
-		"booti 0x02000000 - 0x01f00000\0"
+	"bootcmd=setenv devtype mmc; setenv devnum 1; " \
+		"mmc dev 1; load mmc 1:1 0x00100000 boot.scr; source 0x00100000\0"
 
 #undef CONFIG_BOOTDELAY
-#define CONFIG_BOOTDELAY	1
+#define CONFIG_BOOTDELAY	0
 
 #endif
