--- a/esp_debug.c
+++ b/esp_debug.c
@@ -186,36 +186,36 @@
 
 #else
 
-inline struct dentry *esp_dump_var(const char *name, struct dentry *parent,
+void esp_dump_var(const char *name, struct dentry *parent,
 				   void *value, esp_type type)
 {
-	return NULL;
-}
-
-inline struct dentry *esp_dump_array(const char *name,
+	return;
+}
+
+void esp_dump_array(const char *name,
 				     struct dentry *parent,
 				     struct debugfs_blob_wrapper *blob)
 {
+	return;
+}
+
+void esp_dump(const char *name, struct dentry *parent,
+			       void *data, int size)
+{
+	return;
+}
+
+struct dentry *esp_debugfs_add_sub_dir(const char *name)
+{
 	return NULL;
 }
 
-inline struct dentry *esp_dump(const char *name, struct dentry *parent,
-			       void *data, int size)
-{
-	return NULL;
-}
-
-struct dentry *esp_debugfs_add_sub_dir(const char *name)
-{
-	return NULL;
-}
-
-inline int esp_debugfs_init(void)
+int esp_debugfs_init(void)
 {
 	return -EPERM;
 }
 
-inline void esp_debugfs_exit(void)
+void esp_debugfs_exit(void)
 {
 
 }
--- a/esp_sip.c
+++ b/esp_sip.c
@@ -81,7 +81,7 @@
 #define STRACE_TX_ONE_SHOT_INC() (str.tx_one_shot_overflow++)
 #define STRACE_SHOW(sip)
 #else
-#define esp_sip_dbg(...)
+#define esp_sip_dbg(...) do { } while (0)
 #define STRACE_TX_DATA_INC()
 #define STRACE_TX_CMD_INC()
 #define STRACE_RX_DATA_INC()
--- a/esp_ctrl.c
+++ b/esp_ctrl.c
@@ -451,10 +451,10 @@
 	switch (action_num) {
 	case SIP_AMPDU_RX_START:
 		action->ssn = ssn;
-		// fall through
+		fallthrough;
 	case SIP_AMPDU_RX_STOP:
 		action->index = index;
-		// fall through
+		fallthrough;
 	case SIP_AMPDU_TX_OPERATIONAL:
 	case SIP_AMPDU_TX_STOP:
 		action->win_size = buf_size;
